chef/cookbooks/provisioner/recipes/update_nodes.rb

```diff
@@ -219,19 +222,33 @@ def find_node_boot_mac_addresses(node, admin_data_net)
   ####
   # Everything below is for states that require PXE booting
 
-  append = []
-  mac_list.each_index do |i|
-    dhcp_host "#{mnode.name}-#{i}" do
-      hostname mnode.name
-      macaddress mac_list[i]
-      if admin_mac_addresses.include?(mac_list[i])
-        ipaddress admin_ip_address
-        options [
-          "if exists dhcp-parameter-request-list {
+  if admin_data_net.ip_version == "6"
```

Comment:
This looks to be very much duplicating the next block. I'm not going to block on it, but can comment that it could be refactored to be IP version agnostic.